contracts/button/lib.rs

```diff
@@ -64,18 +52,29 @@ pub mod button_game {
     #[ink(event)]
     #[derive(Debug)]
     pub struct RewardMinted {
-        when: BlockNumber,
+        round: u64,
         #[ink(topic)]
         reward_token: AccountId,
+        #[ink(topic)]
         to: AccountId,
-        amount: Balance,
+        reward: Balance,
+    }
+
+    /// Event emitted when ThePressiah is found at the start of a new round
+    /// triggered by a `reset` tx
+    #[ink(event)]
+    #[derive(Debug)]
+    pub struct PressiahFound {
+        #[ink(topic)]
+        pressiah: AccountId,
+        reward: Balance,
     }
 
     /// Event emitted when the finished game is reset and pressiah is rewarded
     #[ink(event)]
     #[derive(Debug)]
-    pub struct GameReset {
-        when: BlockNumber,
+    pub struct ButtonReset {
+        by: AccountId,
```

Comment:
From what I understand contract should behave in the same way no matter who resets the button. Is it worth it to keep the caller in this event, or maybe we should just have an empty struct here?

Reply:
agreed, makes no sense - removing